torchvision/prototype/features/_feature.py

```diff
@@ -70,12 +71,17 @@ def __torch_function__(
 
         Exceptions to this are:
 
-        - :func:`torch.clone`
+        - :meth:`torch.Tensor.clone`
         - :meth:`torch.Tensor.to`
         """
-        kwargs = kwargs or dict()
+        # Since super().__torch_function__ has no hook to prevent the coercing of the output into the input type, we
+        # need to reimplement the functionality.
+
+        if not all(issubclass(cls, t) for t in types):
+            return NotImplemented
```

Comment:
That was missing in our implementation, but is present in [`super().__torch_function__`](https://github.com/pytorch/pytorch/blob/9a7bfaa92900e6e9c8e2979d4d588ded9d10a1f4/torch/_tensor.py#L1116-L1117).